src/native/libs/System.Native/pal_string.h

```diff
@@ -15,3 +15,12 @@
  * On failure, returns a negative value.
  */
 PALEXPORT int32_t SystemNative_SNPrintF(char* string, int32_t size, const char* format, ...);
+
+/**
+ * Two specialized overloads for use from Interop.Sys, because these two signatures are not equivalent
+ * on some architectures (like 64-bit WebAssembly)
+*/
+
+PALEXPORT int32_t SystemNative_SNPrintF_1S(char* string, int32_t size, const char* format, char* str);
+
+PALEXPORT int32_t SystemNative_SNPrintF_1I(char* string, int32_t size, const char* format, int arg);
```

Comment:
@AaronRobinsonMSFT how did you handle this in runtimelab ?

Reply:
This is the correct way to handle this scenario. Native varargs at the interop boundary have narrow support in .NET and are exclusive to the Windows platform and typically only for the C++/CLI scenario. See https://github.com/dotnet/runtime/issues/48796 for broader support.

The approach taken here is to provide an unmanaged export that managed code can call safely which then forwards to a native varargs function. This ensures the unmanaged calling convention is handled correctly during the P/Invoke and correctly at the dispatch to the unmanaged varargs.

It is ugly and not what anyone really wants, but the cost of supporting native varargs is expensive to support. This comes up rarely relative to other interop scenarios so just hasn't been prioritized. The aforementioned issue mentions some of the costs and relevant parties that would need to respond to support.

Reply:
Makes perfect sense, just wanted to verify.